backend/src/main/java/com/exiger/supplierportal/service/InviteService.java

```diff
@@ -24,41 +27,57 @@ public class InviteService {
     @Autowired
     private ClientRepository clientRepository;
 
+    @Autowired
+    private ClientSupplierRepository clientSupplierRepository;
+
     @Value("${app.frontend.url}")
     private String frontendBaseUrl;
 
     /**
      * Creates a Registration, which generates a one-time temporary registration link for a supplier
      *
-     * @param request InviteRequest containing clientID and supplierEmail
+     * @param request InviteRequest containing clientID, supplierID, and supplierEmail
      * @return InviteResponse containing registrationUrl and expiration
      * @throws IllegalArgumentException if client not found
      */
-    // TODO: update to reflect new registration flow
-//    public InviteResponse createInvite(InviteRequest request) {
-//        // Validate client exists
-//        Client client = clientRepository.findById(request.getClientID())
-//                .orElseThrow(() -> new IllegalArgumentException("Client not found"));
-//
-//        // Generate registration token
-//        UUID token = UUID.randomUUID();
-//        Instant expiration = Instant.now().plus(Duration.ofHours(24)); // 24 hours expiry
-//
-//        Registration registration = new Registration();
-//        registration.setClient(client);
-//        registration.setSupplierEmail(request.getSupplierEmail());
-//        registration.setToken(token);
-//        registration.setExpiration(expiration);
-//
-//        registrationRepository.save(registration);
-//
-//        // Build registration URL
-//        String registrationUrl = frontendBaseUrl + "/register?token=" + token;
-//
-//        InviteResponse response = new InviteResponse();
-//        response.setRegistrationUrl(registrationUrl);
-//        response.setExpiresAt(expiration);
-//
-//        return response;
-//    }
+    public InviteResponse createInvite(InviteRequest request) {
+       // Validate client exists
+       Client client = clientRepository.findById(request.getClientId())
+               .orElseThrow(() -> new IllegalArgumentException("Client not found"));
```

Comment:
```suggestion
               .orElseThrow(() -> new IllegalArgumentException("Client not found with ID: " + request.getClientId()));
```

to match other services that return error when clientId not found